### DISCUSSION TITLE
draw.io diagram's ain't rendering properly, missing elements and other things as well.

### DISCUSSION THREAD

**Discussion by dianTechnoorg** [requarks/wiki#6881] [Error / Bug Report]
# Edit - There are updates
> See the comments for further detail.

# Is there *any* fix or workaround, that I could try?
So, the main issue is what is in the title. Sometimes (especially if there is a lot and big diagrams) it renders completely wrong, missing elements, text boxes, etc...
I have been using Wiki.JS for our internal knowledge system, but we want to use it as an online wiki for our machinery, that we produce and sell, and the draw.io integration is a big deal for me.

> So the question is, what I can do?

 # Why you didn't provide a screenshot?
If needed, I will provide some screenshots, but the current diagrams that are broken have some confidential information, that I am not able to share and/or blur out, because those elements are **THE** broken parts.

# What is the curent structure where the bug is?
There are 3 ` ```diagram` element under each other inside of a `<div>` with a `class=` and right before them there is a `<a name=>` tag too (for easy hyperlinking). All of the  ` ```diagram`s are in a different `<div>`s and all of them have sperate `<a>` tags.

## For example (not the exact .md structure, but close enugh):

``` 
<div class=fig>
   <a name=fig1.1.1></a>

```diagram
<CONTENT OF THE DIAGRAM HERE>```

</div>

<div class=fig>
   <a name=fig1.1.2></a>

```diagram
<CONTENT OF THE DIAGRAM HERE>```

</div>
```
I hope you get the point.
Another thing is, that those missing elements ~only~ in most case happens to go missing when there is an imported image from device (**NOT** linked from another website)

# Some information about the eviroment, that I'm running Wiki.JS
I use wiki.JS in Docker, it have internet connection and many, many custom CSS code **injected** (that could be a problem?), that we are using for branding, and ease of use. 

# Edit: Methods that I tried
I tried to re arrange the diagrams, sometimes it works, sometimes it dosen't that may imply a bigger issue, than I am unaware of, I have tried a fresh new install of Wiki.JS without any of the changes that I made (with the exception of [`master.js` line 94](https://github.com/Requarks/wiki/blob/a1f1cd7356ddc2ba9054aa0f39b0b9f3f43ce3f8/server/master.js#L94), where I increased the maximum file size form 1 mb to 100 mb (according to [this](https://github.com/Requarks/wiki/issues/2797) thread)). and the problem still there (I imported the pages form my github repo).

**Comment by dianTechnoorg** [comment]
# Update no 1:
I am able to workaround this bug, If I delete those problematic elements, and redo those. Sometimes this too ain't helping, but I do it untill it renders properly.

**Comment by dianTechnoorg** [comment]
# Update no. 2
So, it turns out, If I redo the whole diagram, it renders properly. It could be an error with the translation of the diagram to HTML?

Anyway, I managed to fix MY issue, but this bug is still prevalent, so I'm not going to close this.

**Comment by dianTechnoorg** [comment]
# Update no. 3

Okay, so it's been a cuple of days, and there might be an issue with draw.io itself not with Wiki.JS. Maybe with both, IDK, but here is some new info:
The issue seems to be with the fact, that I used <kbd>CTRL</kbd> + <kbd>C</KBD>, <kbd>CTRL</kbd> + <kbd>V</KBD> to copy and paste the text from an already existing `.docx` file (where those diagrams are originally made, and I have to redo it), and those are the affected diagrams. In those, where I typed out, the issues dosen't seems to come up (at time of writing this).

**Comment by jaypoc** [comment]
This just started happening to me on my team's Wiki. We moved from a self hosted Wiki.js to the WIki.js running in a docker container and our Draw.io diagrams are rendering without the text. We can see the shapes/connectors, but the text is blank. When we edit in Draw.io the text is there. The container-hosted version is 2.5.308. I'm looking to reconfigure the old host to see if the issue persists there,

**Reply by scottnursten-22** [reply]
Had the same - found this to work (in Text options)

<img width="406" height="142" alt="screenshot 2025-10-03 at 13 15 15@2x" src="https://github.com/user-attachments/assets/7e9db41e-b396-47e4-bbe8-4016b0d9f995" />

**Reply by jaypoc** [reply]
Thanks, We'll give that a try. 

We also found that just clicking "Edit Page" and then clicking save without making any changes (not even editing/changing the diagram) usually fixed the rendering. This same method also fixed an issue we had where mermaid diagrams were not rending at all on our previous install of WikiJS.

**Reply by jaypoc** [reply]
Just tried that and got strange results.... 

After my fix (just re-rendering the page) it showed up correctly on the wiki, but in the editor I still had a blank diagram (I added a screen shot here.)

When I tried to convert the labels to SVG and saved, in the editor preview, only the boxes in the Yellow section showed labels. The rest were still empty. I went back and re-edited the diagram, thinking I forgot to check some, but confirmed they still had the SVG option checked. I unchecked and re-checked them and saved again, and when It brought be back to the editor, the labels were missing again.

The wiki itself still has the correctly rendered version.

<img width="895" height="454" alt="image" src="https://github.com/user-attachments/assets/947a4f21-46f9-4270-8772-17003f4a28f2" />

**Reply by enricocirignaco** [reply]
> Had the same - found this to work (in Text options)
> 
> <img alt="screenshot 2025-10-03 at 13 15 15@2x" width="406" height="142" src="https://private-user-images.githubusercontent.com/226058422/497126934-7e9db41e-b396-47e4-bbe8-4016b0d9f995.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk1ODY1MjIsIm5iZiI6MTc1OTU4NjIyMiwicGF0aCI6Ii8yMjYwNTg0MjIvNDk3MTI2OTM0LTdlOWRiNDFlLWIzOTYtNDdlNC1iYmU4LTQwMTZiMGQ5Zjk5NS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAwNFQxMzU3MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05M2YyYzEzOGY1NTdhNWUwOTU5ZTI4MDM1NjQxNzZiZjI4YjQ0YzA5NmMwZTM0MGZhZTJiNTM1NWQ1YjlkY2JiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.HoLxAKkDVc5FKUk4EM6TvRC0fGd8QxUGYW0CH2OnCkg">

This worked for me, thanks!
Do I have to do it manually to every diagram or is there a way to apply this setting globally to all diagrams?

**Comment by Pingu-Pox** [comment]
Wanted to share my workaround for this as we saw this too recently.

I selected everything (ctrl+a), and turned off these text properties. Has to be both, not one or the other. Replicated multiple times in different browsers. Issue with text formatting/word wrap rendering?

<img width="241" height="588" alt="image" src="https://github.com/user-attachments/assets/48c24e9a-666e-4460-9d90-6c283508c1d1" />

**Comment by anubi** [comment]
Any insight as to what causes this issue in 2025 to happen would be helpful. It's happening to all my charts on a fresh install on a new server when the same documents worked on a slightly older version of the wiki

**Reply by xterm-inator** [reply]
A fix has now been pushed https://github.com/requarks/wiki/discussions/7744#discussioncomment-15443316
